x-pack/plugins/observability_solution/dataset_quality/public/utils/merge_degraded_docs_into_datastreams.ts

```diff
@@ -12,15 +12,25 @@ export function mergeDegradedStatsIntoDataStreams(
   dataStreamStats: DataStreamStat[],
   degradedDocStats: DegradedDocsStat[]
 ) {
-  const degradedMap: Record<DegradedDocsStat['dataset'], DegradedDocsStat['percentage']> =
-    degradedDocStats.reduce(
-      (degradedMapAcc, { dataset, percentage }) =>
-        Object.assign(degradedMapAcc, { [dataset]: percentage }),
-      {}
-    );
+  const defaultDegradedDocs = {
+    percentage: 0,
+    count: 0,
+  };
+
+  const degradedMap: Record<
+    DegradedDocsStat['dataset'],
+    {
+      percentage: DegradedDocsStat['percentage'];
+      count: DegradedDocsStat['count'];
+    }
+  > = degradedDocStats.reduce(
+    (degradedMapAcc, { dataset, percentage, count }) =>
+      Object.assign(degradedMapAcc, { [dataset]: { percentage, count } }),
+    {}
+  );
 
   return dataStreamStats?.map((dataStream) => ({
     ...dataStream,
-    degradedDocs: degradedMap[dataStream.rawName],
+    degradedDocs: degradedMap[dataStream.rawName] || defaultDegradedDocs,
```

Comment:
Is this necessary? `dataStream.degradedDocs` already has the default values as they are being set in the constructor. Shouldn't this be something like:
```
degradedDocs: degradedMap[dataStream.rawName] || dataStream.degradedDocs,
```
?